src/VisualStudio/Core/Def/Packaging/PackageInstallerServiceFactory.cs

```diff
@@ -364,7 +367,15 @@ private void OnWorkspaceChanged(object sender, WorkspaceChangeEventArgs e)
                 // they've all come in.
                 var cancellationToken = _tokenSource.Token;
                 Task.Delay(TimeSpan.FromSeconds(1), cancellationToken)
-                    .ContinueWith(_ => ProcessBatchedChangesOnForeground(cancellationToken), cancellationToken, TaskContinuationOptions.OnlyOnRanToCompletion, ForegroundTaskScheduler);
+                    .ContinueWith(
+                        async _ =>
+                        {
+                            await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(alwaysYield: true, cancellationToken);
+                            ProcessBatchedChangesOnForeground(cancellationToken);
+                        },
+                        cancellationToken,
+                        TaskContinuationOptions.OnlyOnRanToCompletion | TaskContinuationOptions.ExecuteSynchronously,
+                        TaskScheduler.Default).Unwrap();
```

Comment:
repeat of common pattern.

Reply:
➡️ Covered under discussion in https://github.com/dotnet/roslyn/pull/28781#discussion_r208378244